apps/examples/board_specific_demo/ameba_secure_world_demo/secure_world.c

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+
+#define SHOW_USAGE printf("Usage: secure_world -r -w -a\n \
+		r: run the read test\n \
+		w: run the write test\n \
+		a: take the address as input\n \
+		Valid Range: 0x7034C020 - 0x7039C000\n");
+
+#ifdef  CONFIG_AMEBALITE_TRUSTZONE
+volatile uint32_t* address = 0x735C021;
+#endif
+
+/****************************************************************************
+ * secure_world_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int secure_world_main(int argc, char *argv[])
+#endif
+{
+	int option;
+	volatile int value;
+	while ((option = getopt(argc, argv, "a:rw"))!=-1) {
+		switch (option) {
+		
+		case 'a':
+			sscanf(optarg,"%x",&address);
+			break;
+		case 'r':
+			printf("Trying to read data from %x\n",address);
+			value = *address;
+			break;
+		case 'w':
+			printf("Trying to write data at %x\n",address);
+			*address = 1234; 
+			break;
```

Comment:
this will not work if we give ```secure_world -r -a 0x7....``` as we process -r flag first and it crashes with the default address.  Also, it doesn't check if one of -r or -w flags are given as they are mandatory. So we better set some variables to process the flags and then access the memory. 

Example : 

```
bool r_flag = false;
bool w_flag = false;
.
.
if (!r_flag && !w_flag) SHOW_USAGE
```